### PR TITLE
Add sample app that uses search

### PR DESCRIPTION
**Pull request by kaaes**

### CODE REVIEW

File: sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/CredentialsHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package kaaes.spotify.webapi.samplesearch;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+public class CredentialsHandler {
+
+    public static void setToken(Context context, String token, long expiresIn) {
```
Comment (d-kunin) [inline sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/CredentialsHandler.java]: We could use TimeUnit?

File: sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/CredentialsHandler.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        Context appContext = context.getApplicationContext();
+
+        long now = System.currentTimeMillis();
+        long expiresAt = now + expiresIn * 1000;
+
+        SharedPreferences sharedPref = getSharedPreferences(appContext);
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putString(appContext.getString(R.string.access_token), token);
+        editor.putLong(appContext.getString(R.string.expires_at), expiresAt);
```
Comment (d-kunin) [inline sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/CredentialsHandler.java]: Should we just have constants for keys?

File: sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/SearchPager.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private void getData(String query, int offset, final int limit, final CompleteListener listener) {
+
+        Map<String, Object> options = new HashMap<>();
+        options.put(SpotifyService.OFFSET, offset);
+        options.put(SpotifyService.LIMIT, limit);
+
+        mSpotifyApi.searchTracks(query, options, new SpotifyCallback<TracksPager>() {
+            @Override
+            public void success(TracksPager tracksPager, Response response) {
+//                mHasMoreItems = tracksPager.tracks.total > tracksPager.tracks.offset + tracksPager.tracks.limit;
```
Comment (d-kunin) [inline sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/SearchPager.java]: remove commented code?

File: sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/SearchResultsAdapter.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        Image image = item.album.images.get(0);
+        if (image != null) {
+            Picasso.with(mContext).load(image.url).into(holder.image);
+        }
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return 0;
```
Comment (d-kunin) [inline sample-search/src/main/java/kaaes/spotify/webapi/samplesearch/SearchResultsAdapter.java]: I am not sure, but you might want to return [NO_ID](http://developer.android.com/intl/ru/reference/android/support/v7/widget/RecyclerView.html#NO_ID)

File: sample-search/src/main/res/drawable/item_selector.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <!-- Color when the row is selected -->
+    <item android:drawable="@android:color/darker_gray" android:state_pressed="false" android:state_selected="true" />
+    <!-- Standard background color -->
+    <item android:drawable="@android:color/white" android:state_selected="false" />
+</selector>
```
Comment (d-kunin) [inline sample-search/src/main/res/drawable/item_selector.xml]: This file is never used now?

File: sample-search/src/main/res/drawable/item_selector.xml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <!-- Color when the row is selected -->
+    <item android:drawable="@android:color/darker_gray" android:state_pressed="false" android:state_selected="true" />
+    <!-- Standard background color -->
+    <item android:drawable="@android:color/white" android:state_selected="false" />
```
Comment (d-kunin) [inline sample-search/src/main/res/drawable/item_selector.xml]: For standard case you usually dont need any filters like 'state_selected'.

File: sample-search/src/test/java/com/spotify/sdk/android/ExampleUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+import static org.junit.Assert.*;
+
+/**
+ * To work on unit tests, switch the Test Artifact in the Build Variants view.
+ */
+public class ExampleUnitTest {
+    @Test
+    public void addition_isCorrect() throws Exception {
+        assertEquals(4, 2 + 2);
```
Comment (d-kunin) [inline sample-search/src/test/java/com/spotify/sdk/android/ExampleUnitTest.java]: This is useful test :)

**Comment by d-kunin** [comment]
We could generate some icons assets just to get rid of default one (it is easier then to find the right app on the phone).

**Comment by kaaes** [comment]
@d-kunin good feedback! I made a bunch of changes to address the issues you found

**Pull request merged**